src/main/java/com/powsybl/openloadflow/network/impl/LfBranchImpl.java

```diff
@@ -48,19 +47,63 @@ private static void createLineAsym(Line line, double zb, PiModel piModel, LfBran
                     .setR(piModel.getR())
                     .setX(piModel.getX());
             lfBranch.setAsymLine(new LfAsymLine(piZeroComponent, piPositiveComponent, piNegativeComponent,
-                                                openPhaseA, openPhaseB, openPhaseC));
+                    openPhaseA, openPhaseB, openPhaseC));
         }
     }
 
-    private static LfBranchImpl createLine(Line line, LfNetwork network, LfBus bus1, LfBus bus2, double zb, LfNetworkParameters parameters) {
+    private static LfBranchImpl createLine(Line line, LfNetwork network, LfBus bus1, LfBus bus2, LfNetworkParameters parameters) {
+        double r1;
+        double r;
+        double x;
+        double g1;
+        double b1;
+        double g2;
+        double b2;
+        double zb;
+        if (parameters.getLinePerUnitMode() == LinePerUnitMode.RATIO) {
```

Comment:
Note for @phiedw: @geofjamg and @zamarrenolm have worked on a new way to per-unit lines. The idea is to improve management of lines with different nominal voltage at both ends, especially for initialisation. But, we introduce fatal differences in DC load flow with parameter `setDcUseTransformerRatio(false);` (not better not worst). So we have decided to introduce a parameter `LinePerUnitMode` for your transition. To be removed if everything is okay during next release train.